orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -705,21 +706,24 @@ class GradClippingModel(onnxblock.Model):
         def __init__(self, max_norm):
             self._grad_clip = onnxblock.optim.ClipGradNorm(max_norm)
 
-        def build(self, *grad_names):
-            return self._grad_clip(*grad_names)
+        def build(self, grads_name):
+            return self._grad_clip(grads_name)
 
-    grad_names = []
-    for name, param in pt_model.named_parameters():
-        grad_names.append(f"{name}_grad")
-
-        onnx_model.graph.input.append(
-            onnx.helper.make_tensor_value_info(grad_names[-1], onnx.TensorProto.FLOAT, param.shape)
-        )
+    onnx_model.graph.input.append(
+        onnx.helper.make_tensor_sequence_value_info("gradients", onnx.TensorProto.FLOAT, None)
+    )
 
     grad_clip = GradClippingModel(2.5)
 
+    # Note the call to build() and not to __call__() as onnx does not know
+    # how to infer the output from InplaceClipGradNorm and that will result in
```

Comment:
you mean, the shape infer will just throw, do you think we should have a best effort shape infer, e.g. allowing failed shape infer. 

If we define __call__() is the right function for building graph, we'd better to keep this consistency. What do you think?

Reply:
The problem is that the shape is needed to be known for outputs. So failing shape inference for outputs is not acceptable. We need to somehow populate the output shapes. Let me see if I can update the implementation to populate the shape for the output of this node.